gremlin-client/src/io/serializer_v3.rs

```diff
@@ -112,15 +112,40 @@ where
     if !val.is_empty() {
         let mut x = 0;
         while x < val.len() {
-            let key = get_value!(&val[x], Value::String)?;
+            let key = match &val[x] {
+                Value::String(s) => Ok(s.clone()),
+                Value::Object(_) => {
+                    let token_value: String = match reader(&val[x])? {
+                        GValue::Token(tk) => {
+                            let value = tk.value();
```

Comment:
When I `GValue::Token` is found, we use it's value as string key for the `Map`. This makes it impossible to serialize back to GraphSON using exactly the same structure. We probably need to change the `Map` to support other types as keys. The implementation I suggested here is a middle ground to support the `Token` type.

From the GraphSON docs:
> Map it was necessary to have the ability to return Map instances that did not have String keys (e.g. g.V().out().groupCount()).

So I think we will need to change later to support Int32, etc. But I think this is out of scope for this PR.

Reply:
Hi @rafaelcaricio, `g:T` from docs it' seems a string, am i right?

I agree, we should support other types as keys. 
I've raise an issue for that
https://github.com/wolf4ood/gremlin-rs/issues/10

I will work on this soon since it can change the public api

Reply:
Yes, it's like a string indeed. Sounds good! 👍

Reply:
The difference is that tokens cannot have space, need to start with `a-zA-Z`. So there are some rules that don't apply to string. But in practice it's like a string with restrictions.

Reply:
:+1: